tests/functional/tests.py

```diff
@@ -628,12 +628,23 @@ def test_negative_put_object_with_path_segment(client, log_output):
                           object_name,
                           io.BytesIO(b''), 0)
     except ResponseError as err:
-        pass
+        if err.code != 'XMinioInvalidObjectName':
+            raise Exception(err)
+        else:
+            pass
     except Exception as err:
         raise Exception(err)
     finally:
         try:
-            client.remove_object(bucket_name, object_name)
+            try:
+                client.remove_object(bucket_name, object_name)
+            except ResponseError as err:
+                if err.code != 'XMinioInvalidObjectName':
+                    raise Exception(err)
+                else:
+                    pass
+            except Exception as err:
```

Comment:
lines 646,647 can be removed -since they are just re-raising the exception that would have been raised even without these statements.

Reply:
I am not able to understand... how would that happen?